docs/operating/troubleshooting.md

```diff
@@ -78,3 +78,74 @@ global:
     cluster: eu1
     replica: 0
 ```
+
+# Receiver
+
+## Out-of-bound Error
+
+### Description
+
+#### Thanos Receiver Log
+
+```shell
+level=warn ts=2021-05-01T04:57:12.249429787Z caller=writer.go:100 component=receive component=receive-writer msg="Error on ingesting samples that are too old or are too far into the future" num_droppped=47
+```
+
+### Root Cause
+
+- "Out-of-bound" error occurs when the timestamp of the to-be-written sample is lower than the minimum acceptable timestamp of the TSDB head.
+
+### Possible Cause
+
+1. Thanos Receiver was stopped previously and is just resumed, remote Prometheus starts to write from the oldest sample, which is too old to be digested and hence rejected.
+1. Thanos Receiver does not have enough compute resources to ingest the remote write data (is too slow). The latest ingested sample is gradually falling behind the latest scraped samples.
+
+### Diagnostic and Possible Solution
+
+- Check the pod history of Thanos Receiver to see if it is case #1.
+- For case #2, if you installed Prometheus using the [kube-prometheus-stack](https://github.com/prometheus-community/helm-charts/tree/main/charts/kube-prometheus-stack) helm chart from the Prometheus Community, you can check the "Prometheus / Remote Write dashboard". If the Rate\[5m\] is above 0 for a long period, it is case #2 and you should consider adding replica count or resources to Thanos Receiver.
+
+<img src="../img/thanos_receiver_troubleshoot_grafana_remote_write.png" class="img-fluid" alt="Example Grafana dashboard showing the falling-behind remote write case" />
+
+## Out-of-order Samples Error
+
+### Description
+
+#### Thanos Receiver Log
+
+```shell
+level=warn ts=2021-05-01T05:02:23.596022921Z caller=writer.go:92 component=receive component=receive-writer msg="Error on ingesting out-of-order samples" num_dropped=14
+```
+
+### Root Cause
+
+- TSDB expects to write samples in chronological order for each series.
+- A sample with timestamp t1 is sent to the Thanos Receiver and accepted, any sample with **timestamp t < t1** and **identical label set** being sent to the receiver after this will be determined as out-of-order sample.
+
+### Possible Cause
+
+- Remote Prometheus is running in high availability mode (more than 1 replica are running). But the replica_external_label_name is not correctly configured (e.g. empty).
+
+<img src="../img/thanos_receiver_troubleshoot_empty_replica_external_label_name.png" class="img-fluid" alt="Example topology diagram of out-of-order error case caused by empty replica_external_label_name" />
```

Comment:
BTW, how you created those diagrams? If Google diagrams, do you mind linking the Google drawing and sharing edit rights with Thanos maintainers for easier editions? (:

Reply:
you can use our email `thanos-io@googlegroups.com`

Reply:
> BTW, how you created those diagrams? If Google diagrams, do you mind linking the Google drawing and sharing edit rights with Thanos maintainers for easier editions? (:

I used draw.io, the source files are in xml format. I don't mind sharing those with you guys, maybe I can just sent those source files to the email group in attachment?

Reply:
Maybe add drawio as SVG? WIth https://marketplace.visualstudio.com/items?itemName=hediet.vscode-drawio we can then just IN EDITOR edit svg's. Which contains BOTH the drawio base AND the image itself. 
Pretty..neat with some overhead of-course

Reply:
> Maybe add drawio as SVG? WIth https://marketplace.visualstudio.com/items?itemName=hediet.vscode-drawio we can then just IN EDITOR edit svg's. Which contains BOTH the drawio base AND the image itself.
> Pretty..neat with some overhead of-course

I can save those drawIO files as svg/png (with drawIO source inside, which is its feature) in the thanos repository and refer to them directly in the troubleshoot document, will do it ASAP.

Reply:
added